core/src/main/java/org/neo4j/ogm/cypher/Filter.java

```diff
@@ -220,14 +246,67 @@ public void setNestedRelationshipEntity(boolean nestedRelationshipEntity) {
 		this.nestedRelationshipEntity = nestedRelationshipEntity;
 	}
 
-    /**
-     * Returns the result of passing the property value through the transformer associated with the comparison operator
-     * on this {@link Filter}.
-     *
-     * @return The transformed property value
-     */
-    public Object getTransformedPropertyValue() {
-       return this.comparisonOperator.getPropertyValueTransformer().transformPropertyValue(this.propertyValue);
-    }
+	/**
+	 * Returns the result of passing the property value through the transformer associated with the comparison operator
+	 * on this {@link Filter}.
+	 *
+	 * @return The transformed property value
+	 */
+	public Object getTransformedPropertyValue() {
+		return this.comparisonOperator.getPropertyValueTransformer().transformPropertyValue(this.value);
+	}
+
+	public FilterFunction getFunction() {
+		return function;
+	}
+
+	public void setFunction(FilterFunction function) {
+		this.function = function;
+	}
+
+	/**
+	 * @param nodeIdentifier
+	 * @param addWhereClause
+	 * @return The filter state as a CYPHER fragment.
+	 */
+	public String toCypher(String nodeIdentifier, boolean addWhereClause) {
+		StringBuilder cypher = new StringBuilder();
+		if (this.function == FilterFunction.NONE) {
+			String uniquePropertyName = isNested() ?
+					getNestedPropertyName() + "_" + getPropertyName() : getPropertyName();
```

Comment:
Sounds good - like a version of the 'replace instanceof with polymorphism' refactoring.  You also reminded me - instead of the `DistanceComparison` type I was going to use a Map to supply function parameters. Same type for any future functions. What do you think?